connectors/src/connectors/google_drive/temporal/workflows.ts

```diff
@@ -68,6 +72,25 @@ export async function googleDriveFullSync({
     foldersToBrowse = await getFoldersToSync(connectorId);
   }
 
+  setHandler(folderUpdatesSignal, (folderUpdates: FolderUpdatesSignal[]) => {
+    // If we get a signal, update the workflow state by adding/removing folder ids.
+    for (const { action, folderId } of folderUpdates) {
+      switch (action) {
+        case "added":
+          foldersToBrowse!.push(folderId);
+          break;
+        case "removed":
+          foldersToBrowse!.splice(foldersToBrowse!.indexOf(folderId), 1);
+          break;
+        default:
+          assertNever(
+            "You must handle all values of the variable: action",
```

Comment:
Since `assertNever` serves two purposes here 1. ensuring TypeScript flags the issue 2. it will also be used at runtime in case it gets the wrong signal action. Signals can be sent from the Temporal UI by anyone. Hence, I'd recommend using a more specific error message, such as: "Unexpected signal action received for Google Drive full sync workflow".